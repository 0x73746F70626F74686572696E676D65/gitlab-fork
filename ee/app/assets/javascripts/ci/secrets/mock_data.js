const mockSecretsData = [
  {
    name: 'runner-token-windows',
    labels: [],
    lastAccessed: '2024-03-19T20:55:08.551Z',
    createdAt: '2024-03-11T01:33:06.258Z',
  },
  {
    name: 'product/client-tokens',
    labels: ['env::staging', 'env::production'],
    lastAccessed: '2024-01-23T07:12:44.833Z',
    createdAt: '2024-01-22T08:04:26.024Z',
  },
  {
    name: 'product/server-tokens',
    labels: ['env::staging', 'env::production'],
    lastAccessed: '2024-01-22T07:12:44.833Z',
    createdAt: '2023-01-24T08:04:26.024Z',
  },
  {
    name: 'security-token-project2',
    labels: ['env::production', 'security'],
    lastAccessed: '2024-01-21T23:13:37.227Z',
    createdAt: '2024-01-19T11:59:43.516Z',
  },
  {
    name: 'personal-tokens',
    labels: [],
    lastAccessed: '2024-01-19T21:01:38.925Z',
    createdAt: '2024-01-17T05:51:54.602Z',
  },
  {
    name: 'token-deployment',
    labels: [],
    lastAccessed: '2024-01-16T23:13:31.548Z',
    createdAt: '2024-01-15T17:12:33.882Z',
  },
  {
    name: 'runner-token-mac',
    labels: [],
    lastAccessed: '2024-01-06T07:50:58.308Z',
    createdAt: '2023-01-15T16:35:38.326Z',
  },
  {
    name: 'aws-db-prod-credentials',
    labels: ['AWS'],
    lastAccessed: '2024-01-05T11:47:59.454Z',
    createdAt: '2024-01-04T10:37:41.135Z',
  },
  {
    name: 'runner-token-linux',
    labels: [],
    lastAccessed: '2023-12-30T16:41:08.551Z',
    createdAt: '2023-01-14T18:52:56.258Z',
  },
  {
    name: 'SSH-KEY-1',
    labels: [],
    lastAccessed: '2023-11-29T22:18:31.562Z',
    createdAt: '2023-11-30T01:29:14.394Z',
  },
  {
    name: 'group-ID',
    labels: ['shell'],
    lastAccessed: '2023-11-29T04:30:10.220Z',
    createdAt: '2023-01-21T17:34:33.544Z',
  },
  {
    name: 'security-token-project1',
    labels: ['security'],
    lastAccessed: '2023-11-28T04:30:10.220Z',
    createdAt: '2023-01-21T17:34:33.544Z',
  },
];

export const mockGroupSecretsData = mockSecretsData.map((secret, index) => ({
  key: `group_secret_${index}`,
  ...secret,
}));

export const mockProjectSecretsData = mockSecretsData.map((secret, index) => ({
  key: `project_secret_${index}`,
  ...secret,
}));
