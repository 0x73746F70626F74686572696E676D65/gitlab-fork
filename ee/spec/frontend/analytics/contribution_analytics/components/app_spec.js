import VueApollo from 'vue-apollo';
import Vue from 'vue';
import * as Sentry from '@sentry/browser';
import { shallowMount } from '@vue/test-utils';
import { GlLoadingIcon, GlAlert } from '@gitlab/ui';
import createMockApollo from 'helpers/mock_apollo_helper';
import waitForPromises from 'helpers/wait_for_promises';
import App from 'ee/analytics/contribution_analytics/components/app.vue';
import contributionsQuery from 'ee/analytics/contribution_analytics/graphql/contributions.query.graphql';
import { MOCK_CONTRIBUTIONS_RESPONSE } from '../mock_data';

jest.mock('@sentry/browser');

Vue.use(VueApollo);

describe('Contribution Analytics App', () => {
  let wrapper;

  const mockContributionsHandler = jest.fn();
  const createMockApolloProvider = (contributionsQueryResolver) =>
    createMockApollo([
      [contributionsQuery, mockContributionsHandler.mockResolvedValue(contributionsQueryResolver)],
    ]);

  const createWrapper = ({ mockApollo }) => {
    wrapper = shallowMount(App, {
      apolloProvider: mockApollo,
      propsData: {
        fullPath: 'test',
        startDate: '2000-01-01',
        endDate: '2000-12-31',
      },
    });
  };

  const findLoadingIcon = () => wrapper.findComponent(GlLoadingIcon);
  const findErrorAlert = () => wrapper.findComponent(GlAlert);

  it('renders the loading spinner when the request is pending', async () => {
    const mockApollo = createMockApolloProvider({ data: null });
    createWrapper({ mockApollo });

    expect(findLoadingIcon().exists()).toBe(true);
    await waitForPromises();
    expect(findLoadingIcon().exists()).toBe(false);
  });

  it('renders the error alert if the request fails', async () => {
    const mockApollo = createMockApolloProvider({ data: null });
    createWrapper({ mockApollo });
    await waitForPromises();

    expect(Sentry.captureException).toHaveBeenCalled();
    expect(findErrorAlert().exists()).toBe(true);
    expect(findErrorAlert().text()).toEqual(wrapper.vm.$options.i18n.error);
  });

  it('fetches paginated data', async () => {
    const mockApollo = createMockApolloProvider(MOCK_CONTRIBUTIONS_RESPONSE);
    createWrapper({ mockApollo });
    await waitForPromises();

    expect(mockContributionsHandler).toHaveBeenCalledWith({
      ...wrapper.props(),
      nextPageCursor: '',
    });
  });
});
