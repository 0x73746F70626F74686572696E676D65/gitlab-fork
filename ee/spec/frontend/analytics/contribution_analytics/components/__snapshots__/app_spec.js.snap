// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contribution Analytics App fetches the data per week, using paginated requests when necessary 1`] = `
Array [
  Object {
    "issuesClosed": 175,
    "issuesCreated": 175,
    "mergeRequestsApproved": 175,
    "mergeRequestsClosed": 175,
    "mergeRequestsCreated": 175,
    "mergeRequestsMerged": 175,
    "repoPushed": 175,
    "totalEvents": 175,
    "user": Object {
      "id": "primary",
      "name": "primary",
      "webUrl": "primary",
    },
  },
  Object {
    "issuesClosed": 12,
    "issuesCreated": 12,
    "mergeRequestsApproved": 12,
    "mergeRequestsClosed": 12,
    "mergeRequestsCreated": 12,
    "mergeRequestsMerged": 12,
    "repoPushed": 12,
    "totalEvents": 12,
    "user": Object {
      "id": "secondary",
      "name": "secondary",
      "webUrl": "secondary",
    },
  },
]
`;
